libs/langchain/langchain/llms/llamacpp.py

```diff
@@ -353,6 +390,36 @@ def _stream(
                     token=chunk.text, verbose=self.verbose, log_probs=logprobs
                 )
 
+    async def _astream(
+        self,
+        prompt: str,
+        stop: Optional[List[str]] = None,
+        run_manager: Optional[AsyncCallbackManagerForLLMRun] = None,
+        **kwargs: Any,
+    ) -> AsyncIterator[GenerationChunk]:
+        """Yields results objects as they are generated in real time.
+        It also calls the callback manager's on_llm_new_token event with
+        similar parameters to the OpenAI LLM class method of the same name.
+        Args:
+            prompt: The prompts to pass into the model.
+            stop: Optional list of stop words to use when generating.
+        Returns:
+            An async generator representing the stream of tokens being generated.
+        """
+        params = {**self._get_parameters(stop), **kwargs}
+        result = self.client(prompt=prompt, stream=True, **params)
```

Comment:
This is incorrect, as it would block the event loop for the duration of the streaming request. What should be done here is something like
```py
loop = asyncio.get_running_loop()
iterator = await loop.run_in_executor(None, partial(self.client, prompt=prompt, stream=True, **params))
while True:
  try:
    part = await loop.run_in_executor(next, iterator)
  except StopIteration:
    break
  
  logprobs = part["choices"][0].get("logprobs", None)
  ...
```